crates/nu-parser/src/parser.rs

```diff
@@ -2339,13 +2343,13 @@ pub fn parse_filesize(
 
     let bytes = working_set.get_span_contents(span);
 
+    //todo: parse_filesize_bytes should distinguish between not-that-type and syntax error in units
     match parse_filesize_bytes(bytes, span) {
         Some(expression) => (expression, None),
         None => (
             garbage(span),
-            Some(ParseError::Mismatch(
-                "filesize".into(),
-                "non-filesize unit".into(),
+            Some(ParseError::Expected(
+                "filesize with valid units".into(),
```

Comment:
Do we have an error variant where we could display the valid units as a help text? Or will this `ParseError::Expected` always be hidden through a later parse?

Reply:
::Expected will be hidden in ::Any shape, can't say about other contexts.  I had to change the variant to make parse_filesize() play nice in the ::Any shape.  If it issues any kind of terminating (and therefore user-visible) error, it'll mask what might be a valid bare string, like an executable name.  For example:
```
〉7day
1wk
--------------------------------------------------------------
〉7da
Error: nu::shell::external_command (link)

  × External command failed
   ╭─[entry #181:1:1]
 1 │ 7da
   · ─┬─
   ·  ╰── executable was not found
   ╰────
  help: No such file or directory (os error 2)
```
but if the token were `7zip`, you would not want `parse_duration()` issuing a terminating error.  So we have to figure out how to stop calling many of the entity parsers on what might be the name of an executable.